src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp

```diff
@@ -29,6 +31,27 @@
 
 using namespace mlir;
 
+//===----------------------------------------------------------------------===//
+// Insertion point for initialization instructions and the blocks used for
+// inserting the initialization and main code. These blocks will disappear
+// when the first canonicalization is performed because the init block
+// unconditionally branches into the second block. These blocks exist only for
+// the purpose of this optimization.
+// The support happens on a per function basis.
+//===----------------------------------------------------------------------===//
+
+typedef struct ONNXOperandsInitState {
+  Block *initBlock;
+  Block *mainBlock;
+  BranchOp branchInit;
+  llvm::SetVector<Value> operandsInInitBlock;
+} ONNXOperandsInitState;
+
+typedef std::map<FuncOp, std::unique_ptr<ONNXOperandsInitState>>
+    FunctionToInitStates;
+
+extern std::map<ModuleOp, std::unique_ptr<FunctionToInitStates>> initMap;
```

Comment:
If you prefer adding comments in header file rather than where its actually defined, add comments here instead.

Reply:
I just did please let me know what you think.

Reply:
great, tx